NEW Create Translator (#2)
silverstripe/silverstripe-tx-translator: README.md

# silverstripe tx translator

Module to assist with transifex integration. Works with Silverstripe 4.12+ and 5.0+.

## Operations

- Pull latest translations from transifex and merge them into yml/json/js translation files
- Run i18nTextCollectorTask on a local website
- Push updated source strings to transifex (optional)
- Create github pull-requests with file changes

## Requirements

Transifex cli client (tx) - [https://developers.transifex.com/docs/cli](https://developers.transifex.com/docs/cli) configured API key from transifex.

You must use the go version at least version 1.6+ and not the old python version.

## Usage

> 

Reviewer: ```suggestion

This must be run in the context of a working Silverstripe project (with database, etc). You can install it as a dev dependency within that project if you want, or you can `git clone` it into some directory somewhere - so long as when you run it the working directory in your terminal is at the project root for your Silverstripe project.

```

Author: Not accurate - see comment below in the autoloader

Reviewer: I disagree but apparently I can't convince you so I'll just resolve the conversation.

```
cd /path/to/my-local-site

> composer install silverstripe/tx-translator
> TX_SITE=my-local-site.test TX_PUSH=1 php vendor/silverstripe/tx-translator/script/translate.php

Reviewer: ```suggestion
# if you install this as a dependency of your silverstripe project, `path/to/tx-translator` will be `vendor/silverstripe/tx-translator`
TX_SITE=my-local-site.test TX_PUSH=1 php path/to/tx-translator/script/translate.php
```

Author: Requires `composer install` - see comment below in autoloader section

Reviewer: I disagree but apparently I can't convince you so I'll just resolve the conversation.

```

**To omit the push to transifex**

```

> TX_SITE=mylocalsite.test php vendor/silverstripe/tx-translator/script/translate.php

Reviewer: ```suggestion
TX_SITE=mylocalsite.test php path/to/tx-translator/script/translate.php
```

Author: Not updated because needs to be in vendor folder

Reviewer: I disagree but apparently I can't convince you so I'll just resolve the conversation.

```

## Environment variables

Environment variables can either be set via the command line, like in the examples above, or in an .env in the root folder of the site, or using any other methods to set environment variables.

- `TX_SITE` (required) - the url of a local silverstripe site to run i18nTextCollectorTask against. `http://` will be automatically added if protocol is omitted
- `TX_PULL` (default `1`) - pull latest translations from transifex
- `TX_PUSH` (default `0`) - push new source strings to transifex

> - `DEV_MODE` (default `0`) - do not push to transifex or create github pull-requests. Useful for local development and for doing dry-runs

Reviewer: ```suggestion
- `TX_DEV_MODE` (default `0`) - do not push to transifex or create github pull-requests. Useful for local development and for doing dry-runs
```
Feel free to push back against this one - but since one of the ways to do this is to add the env variable to a silverstripe project's `.env` file I think it's better if this is namespaced so it's clear it's related to the translation stuff

Author: Updated

# silverstripe tx translator

Module to assist with transifex integration. Works with Silverstripe 4.12+ and 5.0+.

## Operations

- Pull latest translations from transifex and merge them into yml/json/js translation files
- Run i18nTextCollectorTask on a local website
- Push updated source strings to transifex (optional)
- Create github pull-requests with file changes

## Requirements

Transifex cli client (tx) - [https://developers.transifex.com/docs/cli](https://developers.transifex.com/docs/cli) configured API key from transifex.

You must use the go version at least version 1.6+ and not the old python version.

> Create a new classic github api token in [github token settings](https://github.com/settings/tokens) with all checkboxes unticked except for `public_repo`. This is required to create pull-requests.

Author: I tried for a long time to get a finely grained token going, but I just couldn't work out the required permissions.  I had 'Actions', 'Contents', 'Pull-request' all set to read & write, still got a response back saying I had the wrong permissions.

Note: there used to be code in here to fallback to use the composer API token - but this is a bad idea, my composer token failed because I lacked permissions. I think saying that you must create a new token with the correct permissions and then delete it in the readme is the best way to reduce confusion for why pull-request don't get created.

Delete this token from github once you have completed updating translations.

## Usage

```
cd /path/to/my-local-site
composer install silverstripe/tx-translator
GITHUB_API_TOKEN=mytoken TX_SITE=my-local-site.test TX_PUSH=1 php vendor/silverstripe/tx-translator/scripts/translate.php
```

**To omit the push to transifex**

```
GITHUB_API_TOKEN=mytoken TX_SITE=my-local-site.test php vendor/silverstripe/tx-translator/scripts/translate.php
```

## Environment variables

> Environment variables can either be set via the command line, like in the examples above, or in an .env in the root folder of the site, or using any other methods to set environment variables.

Reviewer: ```suggestion
Environment variables can either be set via the command line, like in the examples above, or in an .env in the root folder of the site, or using any other methods to set environment variables.

Note that the only valid values for the boolean variables is 1, true, 0, and false.
```

Author: Updated

- `TX_SITE` (required) - the url of a local silverstripe site to run i18nTextCollectorTask against. `http://` will be automatically added if protocol is omitted

> - `TX_PULL` (default `1`) - pull latest translations from transifex

Reviewer: That's not all it does - it's the toggle for _all_ of the updating (tx pull, run the collector task, update js). Please update this to be more accurate.

Author: Updated

- Create github pull-requests with file changes

## Requirements

Transifex cli client (tx) - [https://developers.transifex.com/docs/cli](https://developers.transifex.com/docs/cli) configured API key from transifex.

You must use the go version at least version 1.6+ and not the old python version.

Create a new classic github api token in [github token settings](https://github.com/settings/tokens) with all checkboxes unticked except for `public_repo`. This is required to create pull-requests.

Delete this token from github once you have completed updating translations.

## Usage

```
cd /path/to/my-local-site
composer install silverstripe/tx-translator
TX_GITHUB_API_TOKEN=mytoken TX_SITE=my-local-site.test TX_PUSH=1 php vendor/silverstripe/tx-translator/scripts/translate.php
```

**To omit the push to transifex**

```
TX_GITHUB_API_TOKEN=mytoken TX_SITE=my-local-site.test php vendor/silverstripe/tx-translator/scripts/translate.php
```

## Environment variables

Environment variables can either be set via the command line, like in the examples above, or in an .env in the root folder of the site, or using any other methods to set environment variables.

> Note that the only valid values for the boolean variables is 1, true, 0, and false.

Reviewer: ```suggestion
Note that the only valid values for the boolean variables are "1", "on", "true", "0", "off", and "false".
```
Sorry, should have added quotes the first time around - but I'm only adding them now because a change is needed anyway to add "on" and "off"

Author: You don't actually need to quote them in the .env file, and you definitely don't when passing on cli, so we're probably better off leaving them unquoted

# silverstripe tx translator

Module to assist with transifex integration. Works with Silverstripe 4.12+ and 5.0+.

## Operations

- Pull latest translations from transifex and merge them into yml/json/js translation files
- Run i18nTextCollectorTask on a local website
- Push updated source strings to transifex (optional)
- Create github pull-requests with file changes

## Requirements

Transifex cli client (tx) - [https://developers.transifex.com/docs/cli](https://developers.transifex.com/docs/cli) configured API key from transifex.

You must use the go version at least version 1.6+ and not the old python version.

Create a new classic github api token in [github token settings](https://github.com/settings/tokens) with all checkboxes unticked except for `public_repo`. This is required to create pull-requests.

Delete this token from github once you have completed updating translations.

## Usage

```
cd /path/to/my-local-site

> composer install silverstripe/tx-translator

Reviewer: ```suggestion
composer require silverstripe/tx-translator
```
Sorry, I should have caught this earlier. This is the last thing, then it's all good for merge.

Author: Updated
